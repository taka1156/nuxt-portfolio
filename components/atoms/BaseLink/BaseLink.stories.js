import BaseLink from './BaseLink.vue';

export default {
  title: 'Atoms/BaseLink',
  component: BaseLink
};

const Template = () => ({
  components: { BaseLink },
  template: '<base-link :routeTo="`/`">Text</base-link>'
});

export const Default = Template.bind({});
