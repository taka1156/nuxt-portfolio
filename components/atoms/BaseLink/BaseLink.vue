<template>
  <component
    :is="isInternalLink ? 'nuxt-link' : 'a'"
    :to="isInternalLink ? routeTo : false"
    :href="isInternalLink ? false : routeTo"
    tag="a"
    class="base-link--extend"
    v-bind="isTarget ? { target: '_blank', rel: 'noopener noreferrer' } : false"
  >
    <!-- @slot リンクのクリックに指定したいもの -->
    <slot />
  </component>
</template>

<script>
export default {
  name: 'BaseLink',
  props: {
    /**
     * nuxt-linkの遷移先
     */
    routeTo: {
      type: [String, Object],
      required: true
    },
    isTarget: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    isInternalLink() {
      /**
       * ルーティングか、外部リンクどちらか判定する。
       */
      if (typeof this.routeTo === 'object') {
        return true;
      }
      return `${this.routeTo}`.indexOf('http') === -1;
    }
  }
};
</script>

<style scoped>
a,
.nuxt-link {
  text-decoration: none;
  cursor: pointer;
}
</style>
