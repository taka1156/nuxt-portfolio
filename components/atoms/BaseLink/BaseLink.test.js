import { mount, RouterLinkStub } from '@vue/test-utils';
import {
  dummyOutsideLink,
  dummyRouteStr,
  dummyRouteObj
} from '@/__testdata__/testdata.js';
import BaseLink from './BaseLink.vue';

describe('BaseLink', () => {
  const baseStubLink = propsData =>
    mount(BaseLink, {
      stubs: {
        NuxtLink: RouterLinkStub
      },
      slots: {
        default: 'リンクテキスト'
      },
      propsData: { ...propsData }
    });

  const baseLink = propsData =>
    mount(BaseLink, {
      slots: {
        default: 'リンクテキスト'
      },
      propsData: { ...propsData }
    });

  it('BaseLink初期値(OutSidelink): String', () => {
    const wrapper = baseLink(dummyOutsideLink);
    expect(wrapper.vm.routeTo).toBe(dummyOutsideLink.routeTo);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('BaseLink初期値(NuxtLink): String', () => {
    const wrapper = baseStubLink(dummyRouteStr);
    expect(wrapper.vm.routeTo).toBe(dummyRouteStr.routeTo);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('BaseLink初期値(NuxtLink): Object', () => {
    const wrapper = baseStubLink(dummyRouteObj);
    expect(wrapper.vm.routeTo).toBe(dummyRouteObj.routeTo);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('computed: isInternalLinkのテスト', () => {
    expect(BaseLink.computed.isInternalLink.call(dummyRouteStr)).toBe(true);
    expect(BaseLink.computed.isInternalLink.call(dummyRouteObj)).toBe(true);
    expect(BaseLink.computed.isInternalLink.call(dummyOutsideLink)).toBe(false);
  });
});
