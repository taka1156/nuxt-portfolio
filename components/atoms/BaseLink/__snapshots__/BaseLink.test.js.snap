// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLink BaseLink初期値(NuxtLink): Object 1`] = `"<a class=\\"base-link--extend\\">リンクテキスト</a>"`;

exports[`BaseLink BaseLink初期値(NuxtLink): String 1`] = `"<a class=\\"base-link--extend\\">リンクテキスト</a>"`;

exports[`BaseLink BaseLink初期値(OutSidelink): String 1`] = `"<a href=\\"http://placehold.jp\\" tag=\\"a\\" class=\\"base-link--extend\\">リンクテキスト</a>"`;
